File: auth-server/src/main/java/com/neuqsoft/authserver/repo/UserAuthRepo.java
package com.neuqsoft.authserver.repo;


import com.neuqsoft.authserver.entity.UserAuth;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

/**
 * @author sunjiarui
 * @Date 2020/8/11
 */
@Repository
public interface UserAuthRepo extends JpaRepository<UserAuth, String> {
    /**
     * 按照用户名查询用户
     *
     * @param userName 用户名
     * @return 查询到的用户
     */
    UserAuth findByUserName(String userName);

    /**
     * 按照注册邮箱查询用户
     *
     * @param userEmail 用户邮箱
     * @return 查询到的用户
     */
    UserAuth findByUserEmail(String userEmail);

    /**
     * 按照用户手机号查询用户
     *
     * @param userPhone 用户的手机号
     * @return 查询到的用户
     */
    UserAuth findByUserPhone(String userPhone);
}

File: spring-kotlin-l/src/main/kotlin/org/sheshui/springkotlinl/repo/UserAuthRepo.kt
package org.sheshui.springkotlinl.repo

import org.sheshui.springkotlinl.entity.UserAuth
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux

/**
 * @author sunjiarui
 * @Date 2020/10/12
 */
interface UserAuthRepo : ReactiveCrudRepository<UserAuth, String> {
    /**
     * 查询所有数据
     *
     * @return
     */
    override fun findAll(): Flux<UserAuth>
}